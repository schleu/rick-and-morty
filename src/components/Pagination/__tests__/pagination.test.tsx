import { fireEvent, render } from "@testing-library/react";
import { Pagination } from "..";

import '@testing-library/jest-dom'


describe("Pagination component", () => {
    const mockChangePage = jest.fn();
    const totalPages = 5;
    const actualPage = 2;
      
    it("renders without crashing", () => {
      render(<Pagination actualPage={actualPage} totalPages={totalPages} changePage={mockChangePage} />);
    });
  
    it("snapshot", () => {
      const {container} = render(<Pagination actualPage={actualPage} totalPages={totalPages} changePage={mockChangePage} />);
      expect(container).toMatchSnapshot()
    });
  
    it("displays the current page and total pages", () => {
      const { getByText } = render(<Pagination actualPage={actualPage} totalPages={totalPages} changePage={mockChangePage} />);
      expect(getByText(`${actualPage} de ${totalPages}`)).toBeInTheDocument();
    });
  
    it("calls changePage function with the correct value when 'firstPage' is clicked", () => {
      const { getByText } = render(<Pagination actualPage={actualPage} totalPages={totalPages} changePage={mockChangePage} />);
      fireEvent.click(getByText("<<"));
      expect(mockChangePage).toHaveBeenCalledWith(1);
    });
  
    it("calls changePage function with the correct value when 'previousPage' is clicked", () => {
      const { getByText } = render(<Pagination actualPage={actualPage} totalPages={totalPages} changePage={mockChangePage} />);
      fireEvent.click(getByText("<"));
      expect(mockChangePage).toHaveBeenCalledWith(1);
    });
  
    it("calls changePage function with the correct value when 'nextPage' is clicked", () => {
      const { getByText } = render(<Pagination actualPage={actualPage} totalPages={totalPages} changePage={mockChangePage} />);
      fireEvent.click(getByText(">"));
      expect(mockChangePage).toHaveBeenCalledWith(3);
    });
  
    it("calls changePage function with the correct value when 'lastPage' is clicked", () => {
      const { getByText } = render(<Pagination actualPage={actualPage} totalPages={totalPages} changePage={mockChangePage} />);
      fireEvent.click(getByText(">>"));
      expect(mockChangePage).toHaveBeenCalledWith(totalPages);
    });
  });
  