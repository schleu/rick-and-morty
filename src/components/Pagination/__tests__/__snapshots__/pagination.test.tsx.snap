// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component snapshot 1`] = `
<div>
  <div
    class="pagination"
  >
    <button>
      <span>
        &lt;&lt;
      </span>
    </button>
    <button>
      <span>
        &lt;
      </span>
    </button>
    <p>
      2 de 5
    </p>
    <button>
      <span>
        &gt;
      </span>
    </button>
    <button>
      <span>
        &gt;&gt;
      </span>
    </button>
  </div>
</div>
`;
